src/irmin/store_intf.ml

```diff
@@ -342,12 +342,24 @@ module type S_generic_key = sig
     val pp_hash : t Fmt.t
     (** [pp] is the pretty-printer for commit. Display only the hash. *)
 
-    val v : repo -> info:info -> parents:commit_key list -> tree -> commit Lwt.t
+    val v :
+      ?clear:bool ->
+      repo ->
+      info:info ->
+      parents:commit_key list ->
+      tree ->
+      commit Lwt.t
     (** [v r i ~parents:p t] is the commit [c] such that:
 
         - [info c = i]
         - [parents c = p]
-        - [tree c = t] *)
+        - [tree c = t]
+
+        When [clear] is set (the default), the tree cache is emptied upon the
+        function's completion, mirroring the effect of invoking {!Tree.clear}.
+        When setting [clear=true], subtrees of [t] remain intact: no semantic
+        alterations occur, modifications residing in-memory persist, readily
+        available for future use. *)
```

Comment:
I'm not sure I understand this addition. Should this be:
```suggestion
        When setting [clear=false], subtrees of [t] remain intact: no semantic
        alterations occur, modifications residing in-memory persist, readily
        available for future use. *)
```

Reply:
I've removed the confusing sentence.